### ISSUE TITLE
Error: remote exception: .. Connection reset by peer ..

### ISSUE THREAD

**Issue by leowmjw** [sandstorm-io/sandstorm#513]
Trying out the iPython App but seems there is a "Connection reset" error (as shown below).

Instance at --> https://oasis.sandstorm.io/grain/EuEYcmTBhv2nbWZvfRRMXT
```
Error: Error: remote exception: remote exception: remote exception: ::read(fd, buffer, maxBytes): Connection reset by peer
C++ location:(javascript):??
type: disconnected
```

**Comment by leowmjw** [comment]
Seems to be intermittent; was able to get one instance going before another failed with same error: --> https://oasis.sandstorm.io/grain/wbTSNBzw9HgYdpTB2S47hz

**Comment by kentonv** [comment]
Hi! Thanks for this report. It turned out one of the two worker machines was in a weird state, such that any attempt to start a grain on that machine failed, but the other machine succeeded. Hence every other grain would fail to start up.

I've cleared the bad state, so it's working again now. I've also identified the code which needs to be fixed in Blackrock so that it doesn't happen again.

I will also figure out why my alert (pager) script did not fire here. It should have.
